source/agora/node/BlockSerialize.d

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+
+    Function definition and helper related to serialization
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.node.BlockSerialize;
+
+import agora.common.Data;
+import geod24.bitblob;
+
+/// Type of delegate SerializeDg
+public alias SerializeDg = void delegate(scope const(ubyte)[]) /*pure*/ nothrow @safe;
+
+public ubyte[] serializeFull (T) (scope const auto ref T record)
+    nothrow @trusted
```

Comment:
Why is it `@trusted` ? Also this can be made `pure` (unlike the hashing part)

Reply:
use all pure